codesign/inputparse_test.go

```diff
@@ -0,0 +1,129 @@
+package codesign
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/bitrise-io/go-xcode/v2/autocodesign/certdownloader"
+)
+
+func TestParseCertificates(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   Input
+		want    []certdownloader.CertificateAndPassphrase
+		wantErr bool
+	}{
+		{
+			name: "One certificate and passphrase",
+			input: Input{
+				CertificateURLList:        "https://example.com/storage/development.p12 ",
+				CertificatePassphraseList: "password",
+				KeychainPath:              t.TempDir(),
+				KeychainPassword:          "keychainpassword",
+			},
+			want: []certdownloader.CertificateAndPassphrase{{
+				URL:        "https://example.com/storage/development.p12",
+				Passphrase: "password",
+			}},
+		},
+		{
+			name: "Multiple certificates and passphrases",
+			input: Input{
+				CertificateURLList:        "https://example.com/storage/development.p12|https://example.com/storage/distribution.p12",
+				CertificatePassphraseList: "password|password2",
+				KeychainPath:              t.TempDir(),
+				KeychainPassword:          "keychainpassword",
+			},
+			want: []certdownloader.CertificateAndPassphrase{
+				{
+					URL:        "https://example.com/storage/development.p12",
+					Passphrase: "password",
+				},
+				{
+					URL:        "https://example.com/storage/distribution.p12",
+					Passphrase: "password2",
+				},
+			},
+		},
+		{
+			name: "Empty certificate and passphrase value in lists",
+			input: Input{
+				CertificateURLList:        "https://example.com/storage/development.p12|",
+				CertificatePassphraseList: "password|",
+				KeychainPath:              t.TempDir(),
+				KeychainPassword:          "keychainpassword",
+			},
+			wantErr: true, // Because empty passphrase is a valid value for a no-passphrase-cert, so the list has 2 items
+		},
+		{
+			name: "No certificate nor passphrase",
```

Comment:
We may want to allow this in the future (Xcode Cloud Signing can use distribution certs without them being specified here)

Reply:
Fair point, I didn't think about that. But if we decide to support Cloud Signing in every step, it's going to require many other changes as well, for example, every step should allow an empty certificate URL too, not just an empty passphrase.